src/daemon/rpmostreed-transaction-types.cxx

```diff
@@ -1274,25 +1274,55 @@ deploy_transaction_execute (RpmostreedTransaction *transaction,
           return glnx_throw (error, "No overrides for package '%s'", name_or_nevra);
         }
 
-      changed = TRUE;
-    }
-
-  if (self->override_replace_local_pkgs)
-    {
-      g_autoptr(GPtrArray) pkgs = NULL;
-      if (!import_many_local_rpms (repo, self->override_replace_local_pkgs, &pkgs,
-                                   cancellable, error))
-        return FALSE;
-
-      if (pkgs->len > 0)
+      if (self->override_replace_local_pkgs)
         {
-          g_ptr_array_add (pkgs, NULL);
-          if (!rpmostree_origin_add_overrides (origin, (char**)pkgs->pdata,
-                                               RPMOSTREE_ORIGIN_OVERRIDE_REPLACE_LOCAL,
-                                               error))
+          g_autoptr(GPtrArray) pkgs = NULL;
+          if (!import_many_local_rpms (repo, self->override_replace_local_pkgs, &pkgs,
+                                       cancellable, error))
             return FALSE;
-          changed = TRUE;
-        }
+
+          for (char **it = (char**)pkgs->pdata; it && *it; it++)
+            {
+              const char *name_or_nevra = *it;
```

Comment:
Cool, this is going in the right direction! :)

So the thing here is that those strings are actually in the format `<sha256>:NEVRA`. But the only bit we want is `N` (i.e. the package name). I think eventually, it'd be cleaner to just have `import_many_local_rpms` return that, but for now you can use `rpmostree_decompose_sha256_nevra` to extract the NEVRA, and then `rpmostree_decompose_nevra` to extract the name from that. You can `git grep` the codebase to see how those functions are used elsewhere.